src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -175,6 +183,12 @@ func (s *commitLogSource) Read(
 	ctx, span, _ := ctx.StartSampledTraceSpan(tracepoint.BootstrapperCommitLogSourceRead)
 	defer span.Finish()
 
+	// bail early if this source already ran before.
+	if s.bootstrapResults.Results != nil {
+		s.log.Info("the entire range of the commit has already been read. returning previous results")
+		return s.bootstrapResults, nil
+	}
```

Comment:
Can we do this before the span is started? Currently we'll have a lot of 0 length spans if the cache is hit